### PR TITLE
Buildbot publish

### PR DESCRIPTION
**Pull request by pktpls**
This is a bunch of little fixes, and publishing of targets/packages builds to the respective *-snapshot directories. Also includes #62.

### CODE REVIEW

File: roles/buildbot/files/packages.py [hunk 1]
```diff
@@ -46,10 +45,10 @@ def packagesConfig(c):
...
 def archTriggerStep(arch):
   return AsyncTrigger(
-    # here is the possiblibilty of running into a nasty bug. Apparently, names
-    # for virt-builders shouldn't get too long. otherwise they might not get spawned
-    # https://github.com/buildbot/buildbot/issues/3413
-    name=util.Interpolate("trigger p/%(prop:branch)s/%(kw:arch)s", arch=arch),
+    # Step name limit is 50 chars, longest is currently 48 chars:
+    # packages/openwrt-22.03/arm_cortex-a15_neon-vfpv4
+    # See https://github.com/buildbot/buildbot/issues/3413
+    name=util.Interpolate("packages/%(prop:branch)s/%(kw:arch)s", arch=arch),
```
Comment (Akira25) [inline roles/buildbot/files/packages.py:51]: Did you check against openwrt21.02 too? I think there was one target with an even longer name.

File: roles/buildbot/files/symlink_build.sh [hunk 2]
```diff
@@ -1,56 +0,0 @@
-#!/bin/sh
```
Comment (Akira25) [inline roles/buildbot/files/symlink_build.sh:1]: Let us please keep that skript. It really makes things easy, if we have to repair things manually. For example, if the buildbot would be down for some time and we must change to old artifacts

File: roles/buildbot/files/packages.py [hunk 3]
```diff
@@ -114,10 +116,23 @@ def packagesFactory(f):
...
-            command=["true"]))
+        steps.MasterShellCommand(
+            name="publish",
+            haltOnFailure=True,
+            command=["sh", "-c", util.Interpolate("""\
+mkdir -p %(kw:p)s %(kw:p)s.new \
+    && cp -a %(kw:w)s/* %(kw:p)s.new/ \
+    && mv %(kw:p)s %(kw:p)s.prev \
+    && mv %(kw:p)s.new %(kw:p)s \
+    && rm -rf %(kw:p)s.prev \
```
Comment (Akira25) [inline roles/buildbot/files/packages.py:133]: Could you please explain a bit further, how this works? I'm a bit confused by the moves.

Wouldn't be having a symlink a more efficient solution, mostly in terms of storage-efficiency?

**Review by Akira25 — COMMENTED** [review]
In overall looks good to me, with some questions left. Please have a look on them

File: roles/buildbot/files/packages.py (hunk 1 above)
Comment (pktpls) [reply roles/buildbot/files/packages.py:51]: I did in fact, this is the longest one :-)

**Comment by pktpls** [comment]
This should be good now - we now have complete successful snapshot builds for packages and targets!

- [x] Targets builds for Falter branches: snapshot, 1.3.0-snapshot, 1.2.3-snapshot
- [x] Packages builds for OpenWrt branches: master, openwrt-22.03, openwrt-21.02

Now the last big thing missing is stable releases.

**Review by Akira25 — APPROVED** [review]
Looks good to me. Thank you a lot for all the effort spent on this important part of infrastructure. :)

**Pull request merged**